packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -48,6 +78,49 @@ export function cloudflare<T extends Record<string, WorkerOptions>>(
 		},
 		configResolved(resolvedConfig) {
 			viteConfig = resolvedConfig;
+			requiresNodeCompat = getNodeCompatEnvironments(
+				pluginConfig,
+				path.resolve(
+					path.dirname(viteConfig.configFile ?? './dummy'),
+					viteConfig.root,
+				),
+			);
+		},
+		resolveId(source) {
+			if (source.startsWith(CLOUDFLARE_VIRTUAL_PREFIX)) {
+				// We found a potential Node.js import.
+				// If Node.js compatibility is turned on, use the alias to the appropriate unenv polyfill.
+				// Otherwise return it to the original non-aliased import.
+				const [to, from] = source
+					.slice(CLOUDFLARE_VIRTUAL_PREFIX.length)
+					.split('?');
+				if (requiresNodeCompat.has(this.environment.name)) {
+					return to;
+				} else {
+					return from;
+				}
+			}
+		},
+		transform(code, id) {
+			// If the current environment needs Node.js compatibility, then inject the necessary global polyfills.
+			if (requiresNodeCompat.has(this.environment.name)) {
+				// TODO: check whether there is a better way to only do this on entry-points
```

Comment:
`resolveId` has an `isEntry` option passed to it.
But for some reason that hook was not being triggered, so I couldn't capture that.